### PR TITLE
group operations support

### PR DESCRIPTION
**Pull request by bergano65**

### CODE REVIEW

File: CogsMinimizer/Views/Subscription/Analyze.cshtml [hunk 1]
```diff
@@ -571,10 +623,34 @@
...
         </div>
 
-        <div class="col-md-6 pull-right" style="width:250px">
+        <div class="col-md-d6 pull-left" style="width:300px">
+
+            <div class="panel panel-default">
+                <div style="text-align:left" class="panel-heading">Extend all resources in group</div>
+                <div class="panel-body">
+                    <div class="row" style="align-content:left;margin-left:0px;margin-right:20px;padding-bottom:10px;">
+                        @Html.DropDownList("groupList", Model.GroupList, "Select group")
```
Comment (shklar) [inline CogsMinimizer/Views/Subscription/Analyze.cshtml]: Please change the text to "Select resource group"

File: CogsMinimizer/Views/Subscription/Analyze.cshtml [hunk 2]
```diff
@@ -571,10 +623,34 @@
...
+
+                    </div>
+                    <div style="align-content:center" class="row">
+                        <button onclick="ResourceGroupOp('extend')" style="margin-left:15px" class="btn btn-info btn-xs glyphicon glyphicon-play tbl-btn"></button>&nbsp;Extend resources
+                    </div>
+                    <div class="row">
+                        <button onclick="ResourceGroupOp('reserve')" style="margin-left:15px" class="btn btn-success btn-xs glyphicon glyphicon-forward tbl-btn"></button>&nbsp;Reserve resources
+                    </div>
+                    <div class="row">
+                        <button onclick="ResourceGroupOp('reset')" style="margin-left:15px" class="btn btn-default btn-xs glyphicon glyphicon-step-backward tbl-btn"></button>&nbsp;Reset resources
```
Comment (shklar) [inline CogsMinimizer/Views/Subscription/Analyze.cshtml]: Please remove the reset operation

File: CogsMinimizer/Controllers/SubscriptionController.cs [hunk 3]
```diff
@@ -19,6 +19,76 @@ public class SubscriptionController : SubMinimizerController
...
+            {
+                var subscription = db.Subscriptions.FirstOrDefault(x => x.Id.Equals(SubscriptionId));
+
+                if (subscription == null)
+                {
+                    throw new ArgumentException(string.Format("Subscription with ID '{0}' wasn't found.", SubscriptionId));
+                }
+
+                string currentUser = AzureAuthUtils.GetSignedInUserUniqueName();
+                if (currentUser != subscription.ConnectedBy)
```
[the remark this answers is not in the document]
Comment (bergano65) [reply CogsMinimizer/Controllers/SubscriptionController.cs]: fixed

File: CogsMinimizer/Controllers/SubscriptionController.cs [hunk 4]
```diff
@@ -19,6 +19,76 @@ public class SubscriptionController : SubMinimizerController
...
+                            resource.Status = ResourceStatus.Valid;
+                            break;
+                        case "reserve":
+                            // Reserve resources of given  group and subscription
+                            resource.Owner = AzureAuthUtils.GetSignedInUserUniqueName();
+                            resource.ConfirmedOwner = true;
+                            resource.ExpirationDate = ResourceOperationsUtil.GetNewReserveDate(subscription, resource);
+                            resource.Status = ResourceStatus.Valid;
+                            break;
+                        case "reset":
```
[the remark this answers is not in the document]
Comment (bergano65) [reply CogsMinimizer/Controllers/SubscriptionController.cs]: fixed

File: CogsMinimizer/Views/Subscription/Analyze.cshtml (hunk 1 above)
Comment (bergano65) [reply CogsMinimizer/Views/Subscription/Analyze.cshtml]: fixed

File: CogsMinimizer/Views/Subscription/Analyze.cshtml (hunk 2 above)
Comment (bergano65) [reply CogsMinimizer/Views/Subscription/Analyze.cshtml]: fixed

**Pull request merged**